packages/ra-ui-materialui/src/list/List.js

```diff
@@ -33,6 +33,7 @@ export const styles = {
         justifyContent: 'space-between',
         alignSelf: 'flex-start',
     },
+    children: { display: 'flex' },
```

Comment:
You're breaking BC by adding a new flex style. If you just want to make it configurable, you just need to have an empty object literal here.